### ISSUE TITLE
Using fixed set of pos/neg pairs?

### ISSUE THREAD

**Issue by zabzug-pfpt** [KevinMusgrave/pytorch-metric-learning#259]
I see the [sampler to generate a fixed set of triplets](https://kevinmusgrave.github.io/pytorch-metric-learning/samplers/#fixedsetoftriplets) and the [miner that uses that fixed set](https://kevinmusgrave.github.io/pytorch-metric-learning/miners/#embeddingsalreadypackagedastriplets). Is there an analogous sampler/miner duo for generating and using a fixed set of positive and negative pairs?

Also, is there any relevant example code?

**Comment by KevinMusgrave** [comment]
That doesn't exist currently. But pull requests are welcome! 😄

**Comment by zabzug-pfpt** [comment]
@KevinMusgrave it looks like the `EmbeddingsAlreadyPackagedAsTriplets` miner will trivially fail for batch sizes that aren't multiples of 3:
```{python}
        batch_size = embeddings.size(0)
        a = torch.arange(0, batch_size, 3)
        p = torch.arange(1, batch_size, 3)
        n = torch.arange(2, batch_size, 3)
```
From the `BasePostGradientMiner`'s output assertion method:
```{python}
            self.num_triplets = len(output[0])
            assert self.num_triplets == len(output[1]) == len(output[2])
```
For example, with a batch size of 32, we get `len(a) == len(p) == 11` but `len(n) == 10`. Is that expected? Am I missing something?

**Comment by KevinMusgrave** [comment]
Yeah the batch size has to be a multiple of 3. Using the ```FixedSetOfTriplets``` sampler should make the dataloader return triplets arranged as ```[anchor1, positive1, negative1, anchor2, positive2, negative2, ...]```. If the batch size is not a multiple of 3, then you'll be missing a part of a triplet.
